gpiozero/spi_devices.py

```diff
@@ -72,10 +111,10 @@ class AnalogInputDevice(SPIDevice):
     .. _analog to digital converters: https://en.wikipedia.org/wiki/Analog-to-digital_converter
     """
 
-    def __init__(self, bits=None, **spi_args):
-        if bits is None:
-            raise InputDeviceError('you must specify the bit resolution of the device')
+    def __init__(self, bits, **spi_args):
         self._bits = bits
+        self._min_value = -(2 ** bits)
+        self._range = 2 ** (bits + 1) - 1
```

Comment:
Ahh, not quite. For the pseudo-differential devices, they actually operate 0..2**bits just like in single mode. They're called pseudo-differential because they can't do <0 (where IN- is greater than IN+).

For the full-differential ADCs they operate in, e.g. 12-bits in single-ended mode, and 13-bits (2's complement) in full-different mode. So, for the sake of ease, we treat them as 12-bit whatever mode they're in and assume that their range is -2**bits .. (2**bits) - 1. When they're in single ended mode this is fine because the output will never be <0 (it'll just occupy the top half of the range). In full differential mode, this is also fine as it's the full range. Hence, the full numeric range of their output is 2 ** (bits + 1) - 1 values.